ckanext/xloader/plugin.py

```diff
@@ -231,17 +230,14 @@ def get_helpers(self):
         }
 
 
-def _should_remove_unsupported_resource_from_datastore(res_dict_or_obj):
+def _should_remove_unsupported_resource_from_datastore(res_dict):
     if not toolkit.asbool(toolkit.config.get('ckanext.xloader.clean_datastore_tables', False)):
         return False
-    if isinstance(res_dict_or_obj, Resource):
-        res_dict_or_obj = res_dict_or_obj.__dict__
-    return ((not XLoaderFormats.is_it_an_xloader_format(res_dict_or_obj.get('format', u''))
-                or res_dict_or_obj.get('url_changed', False))
-            and (res_dict_or_obj.get('url_type') == 'upload'
-                or res_dict_or_obj.get('url_type') == '')
-            and (res_dict_or_obj.get('datastore_active', False)
-                or res_dict_or_obj.get('extras', {}).get('datastore_active', False)))
+    return (not XLoaderFormats.is_it_an_xloader_format(res_dict.get('format', u''))
+            and (res_dict.get('url_type') == 'upload'
+                or res_dict.get('url_type') == '')
```

Comment:
This should allow any falsy value. We've been tripped up by that before.

Reply:
Oh like if the `datastore_active` is 0 or 1 instead of True/False?

Reply:
No, the url_type can be `None` instead of an empty string.

Reply:
Okay this should be done now, the asbool helper method raise an error so had to deal with that